### PR TITLE
Add tests for dataSize of -1 (main)

### PR DESCRIPTION
**Pull request by alanking**
In service of #7154, #7160

### CODE REVIEW

**Comment by alanking** [comment]
Oh, confirmed that the tests failed appropriately when run after reverting b0c3a687553ff62330e3b901003e5da8e80bea24.

File: scripts/irods/test/test_resource_types.py [hunk 1]
```diff
@@ -532,6 +532,31 @@ def acPostProcForParallelTransferReceived(rule_args, callback, rei):
         except:
             print('Skipping for plugin name ['+self.plugin_name+']')
 
+    def test_unixfilesystem_freespace_with_istream__issue_7160(self):
```
Comment (korydraughn) [inline scripts/irods/test/test_resource_types.py:535]: Can we add a couple more words to the test name to describe what it's about?

It's not clear to me what the goal of the test is.

It seems the test is proving `istream` works when the freespace options are enabled on a resource. Is that the goal of the test?

Comment (alanking) [reply scripts/irods/test/test_resource_types.py:535]: What about...
1. `test_istream_works_with_freespace_and_minimum_free_space_for_create_in_bytes_configured__issue_7160`
2. `test_istream_does_not_fail_with_freespace_and_minimum_free_space_for_create_in_bytes_configured__issue_7160`
3. `test_istream_works_with_minimum_free_space_for_create_in_bytes_configured__issue_7160`
4. `test_istream_does_not_fail_with_freespace_and_minimum_free_space_for_create_in_bytes_configured__issue_7160`

Comment (korydraughn) [reply scripts/irods/test/test_resource_types.py:535]: (1) and (3) work the best to me. They are both clear IMO.

Thoughts?

Comment (alanking) [reply scripts/irods/test/test_resource_types.py:535]: I vote 3. Thanks!

**Comment by alanking** [comment]
These tests both pass, but running core tests just in case. Unit tests all passed

**Review by korydraughn — COMMENTED** [review]
No obvious issues. Please squash to taste without pounds.

**Comment by alanking** [comment]
I went ahead and squashed. Still waiting on core tests, but they'll complete in a few minutes. So far so good.

**Review by korydraughn — APPROVED** [review]
Pound/Merge once tests pass.

**Pull request merged**